service/controller/resource/generic/create.go

```diff
@@ -11,6 +11,9 @@ import (
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	if r.deleteIfMimirEnabled {
+		return r.EnsureDeleted(ctx, obj)
+	}
```

Comment:
I'm not sure to understand the logic behind this. Could you explain pls ? :pray:

Reply:
Of course, so the idea is that if we enabled this property on the controller resource, then we will Always delete instead of create. The goal is that we ensure we delete all existing resources managed by pmo when Mimir is enabled on the installation. This is a conscious choices per resource déclaration

Reply:
Oh ok I think I get it :+1:

Reply:
So the goal is to mimick this behavior https://github.com/giantswarm/prometheus-meta-operator/blob/f26129377cacc9c48179e6efcbf895d10859f804/service/controller/resource/alerting/heartbeatwebhookconfig/create.go#L14 for ressources built using the generic resource. This then avoid the risk of moving away from it

Reply:
Once the resource is deleted, next reconciliation, we are going to enter again in that EnsureCreated and again in EnsureDeleted in case of  Mimir enabled, and we will encounter an error in the EnsureDeleted function as the object won't be there anymore, no ?

Reply:
Nope https://github.com/giantswarm/prometheus-meta-operator/blob/435ec6573943b9440ea4d547d94003e029dadfbd/service/controller/resource/generic/delete.go#L20C2-L24C3